mapping_parenting_tech/analysis/prototyping/app_store/app_store_reviews.py

```diff
@@ -0,0 +1,193 @@
+#                 - `description`
+#                     - `standard` (str) (long description of the app)
+#                 - `kind` (str) (e.g., 'iosSoftware')
+#                 - `name` (str) (e.g., 'Baby games for one year olds.')
+#                 - `subtitle` (str) (e.g., 'Learning for kids age 1 2 3 4')
+# - `pageData`
+#     - `customersAlsoBoughtApps`
+#         - [array of ids of other apps] (int)
+#     - `moreByThisDeveloper`
+#         - [array of ids of other apps] (int)
+#     - `customerReviewsUrl` (str) (URL of customer reviews)
+# - `properties` (just two items, neither seem relevant - 'revNum' and timestamp)
+
+# %%
+url = "http://itunes.apple.com/app/id" + str(app_id)
+res = requests.get(url, headers=headers)
+
+app_full_details = json.loads(res.text)
+app_key_details = {
+    **app_full_details["storePlatformData"]["product-dv"]["results"][f"{app_id}"],
+    **app_full_details["pageData"],
+}
+app_description = app_key_details["description"]["standard"]
+app_reviews_url = app_key_details["customerReviewsUrl"]
+# note that this URL points to a summary page about the app's reviews; see next cell
+
+# %% [markdown]
+# Get information about an app's reviews (namely, how many), but not specific reviews
+
+# %%
+base = "http://itunes.apple.com/WebObjects/MZStore.woa/wa/customerReviews?"
+# NB seems equivalent to https://itunes.apple.com/gb/customer-reviews/id{app_id}?dataOnly=true&displayable-kind=11
+
+params = {"id": app_id, "displayable-kind": 11, "sort": 1}
+url = base + urllib.parse.urlencode(params)
+res = requests.get(url, headers=headers)
+if res.status_code == 200:
+    print("Success gathering review info")
+    reviewInfo = json.loads(res.text)
+else:
+    sys.exit("Fail")
+
+totalReviews = reviewInfo["totalNumberOfReviews"]
+
+# %% [markdown]
+# Get app's reviews
+
+# %%
+base = "http://itunes.apple.com/WebObjects/MZStore.woa/wa/userReviewsRow?"  # this gets reviews
+params = {
+    "id": app_id,
+    "displayable-kind": 11,
+    "startIndex": 0,
+    "endIndex": totalReviews,
+    "sort": 1,
+}
+url = base + urllib.parse.urlencode(params)
+
+res = requests.get(url, headers=headers)
+
+if res.status_code == 200:
+    print("Success gathering reviews")
+    reviews = json.loads(res.text)
+else:
+    sys.exit("Fail")
+
+pd = pandas.DataFrame(reviews["userReviewList"])
+
+print(len(reviews["userReviewList"]))
+# for review in reviews['userReviewList']:
+#    print(review['body'])
+
+pd.sample(10)
+
+# %% [markdown]
+# Code below is obsolete / testing code
+
+# %%
+from mapping_parenting_tech.analysis.prototyping.app_store.api.itunes_app_scraper.scraper import (
+    AppStoreScraper,
+)
+
+scraper = AppStoreScraper()
+results = scraper.get_app_ids_for_query("fortnite", country="gb", lang="en")
+similar = scraper.get_similar_app_ids_for_app(results[0])
+
+app_details = scraper.get_multiple_app_details(similar)
+# print(list(app_details))
+
+# %%
+from api.itunes_app_scraper.scraper import AppStoreScraper
```

Comment:
Threw me an error here: `No module named 'api'`

Reply:
Yes - sorry, from initial testing that I stopped working on when I brought into this repo. Will comment out for now.